### ISSUE TITLE
[2017.7.0rc1] issues with mine.get

### ISSUE THREAD

**Issue by sjorge** [saltstack/salt#42191]
### Description of Issue/Question
My minions throw exceptions on mine.get after upgrading
```
    2017-07-07 01:19:05,308 [salt.state                                                             ][ERROR   ][27943] An exception occurred in this state: Traceback (most recent call last):
      File "/opt/local/lib/python2.7/site-packages/salt/state.py", line 1828, in call
        **cdata['kwargs'])
      File "/opt/local/lib/python2.7/site-packages/salt/loader.py", line 1782, in wrapper
        return f(*args, **kwargs)
      File "/opt/local/lib/python2.7/site-packages/salt/states/x509.py", line 494, in certificate_managed
        new = __salt__['x509.create_certificate'](testrun=True, **kwargs)
      File "/opt/local/lib/python2.7/site-packages/salt/modules/x509.py", line 1379, in create_certificate
        arg=str(kwargs))[ca_server]
    KeyError: 'cronos'
```

### Setup
snippet form state file
```yaml
# ca_host is cronos for my env
{% set ca_crt = salt['mine.get'](ca_host, 'x509.get_pem_entries') %}
```

snippet from state that creates the ca
```yaml
mine.send:
  module.run:
    - func: x509.get_pem_entries
    - kwargs:
        glob_path: {{ certcfg['config']['authority_dir'] }}/ca.crt
    - onchanges:
      - x509: certificate.authority::certificate
```

### Steps to Reproduce Issue
A slightly customized setup as demoed here: https://docs.saltstack.com/en/develop/ref/states/all/salt.states.x509.html#module-salt.states.x509

PS: this is awesome BTW and has been purring along since 2016.3 for me without issue, until now.

### Versions Report
```
Salt Version:
           Salt: 2017.7.0rc1

Dependency Versions:
           cffi: 1.9.1
       cherrypy: 8.1.0
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: 1.6.7
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: 1.5.0
       M2Crypto: 0.22
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec 31 2016, 10:50:46)
   python-gnupg: 2.2.0
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: 0.6.6
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.3
            ZMQ: 4.1.4

System Versions:
           dist:
         locale: UTF-8
        machine: i86pc
        release: 5.11
         system: SunOS
        version: Not Installed
```

**Comment by sjorge** [comment]
I did verify that the correct data is indeed in ```/var/cache/salt/master/minions/cronos/mine.p``` which is were all the mine data is store if I am not mistaken.

**Comment by sjorge** [comment]
This might be related to #42148, mine.send/mine.get seem to be broken if that parameter is specified.

It works fine once removed.

**Comment by gtmanfred** [comment]
If you update your mine.send function to 

```
send mine:
  module.run:
    - mine.send:
      - func: x509.get_pem_entries
      - glob_path: {{ certcfg['config']['authority_dir'] }}/ca.crt
    - onchanges:
      - x509: certificate.authority::certificate
```

Does this work?

**Comment by gtmanfred** [comment]
Here is an example.

```
[root@salt salt]# salt-call mine.get \* test.echo
local:
    ----------
[root@salt salt]# salt-call state.apply test
local:
----------
          ID: test echo
    Function: module.run
      Result: True
     Comment: mine.send: True
     Started: 14:18:39.939795
    Duration: 511.205 ms
     Changes:
              ----------
              mine.send:
                  True

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 511.205 ms
[root@salt salt]# salt-call mine.get \* test.echo
local:
    ----------
    salt.c.gtmanfred-1263.internal:
        hi!
[root@salt salt]# cat /srv/salt/test.sls
test echo:
  module.run:
    - mine.send:
      - func: test.echo
      - text: hi!
```

**Comment by sjorge** [comment]
Fixed after following #42148, I'll close this one.
